packages/plugin-commands-outdated/test/index.ts

```diff
@@ -190,6 +190,65 @@ https://github.com/kevva/is-positive#readme
   }
 })
 
+test('pnpm outdated: format json', async () => {
+  tempDir()
+
+  await fs.mkdir(path.resolve('node_modules/.pnpm'), { recursive: true })
+  await fs.copyFile(path.join(hasOutdatedDepsFixture, 'node_modules/.pnpm/lock.yaml'), path.resolve('node_modules/.pnpm/lock.yaml'))
+  await fs.copyFile(path.join(hasOutdatedDepsFixture, 'package.json'), path.resolve('package.json'))
+
+  {
+    const { output, exitCode } = await outdated.handler({
+      ...OUTDATED_OPTIONS,
+      dir: process.cwd(),
+      format: 'json',
+    })
+
+    expect(exitCode).toBe(1)
+    expect(stripAnsi(output)).toBe(`{
+\t"@pnpm.e2e/deprecated": {
+\t\t"current": "1.0.0",
+\t\t"latest": "Deprecated"
+\t},
+\t"is-negative": {
+\t\t"current": "1.0.0",
+\t\t"latest": "2.1.0"
+\t},
+\t"is-positive": {
+\t\t"current": "1.0.0",
+\t\t"latest": "3.1.0",
+\t\t"dependencyKind": "dev"
+\t}
+}`)
```

Comment:
it looks pretty ugly. You could just run `JSON.stringify` to generate the expected output.

Reply:
good idea! Thank you! [2f3f4f5](https://github.com/pnpm/pnpm/pull/5582/commits/2f3f4f5e82ce83f1ce24db2ca551862cc6977ff1) 👍